File: src/backend/php/app/Http/Requests/UpdateTodoItemRequest.php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;

/**
 * Validates incoming update requests (all fields optional — PATCH semantics).
 * Analogous to an UpdateTodoItemRequest DTO with nullable properties in C#.
 */
class UpdateTodoItemRequest extends FormRequest
{
    public function authorize(): bool
    {
        return true;
    }

    public function rules(): array
    {
        return [
            'title'        => ['sometimes', 'filled', 'string', 'max:200'],
            'description'  => ['sometimes', 'nullable', 'string', 'max:2000'],
            'is_completed' => ['sometimes', 'boolean'],
        ];
    }
}

File: src/backend/java/todo-shared/src/main/java/com/example/todo/dto/UpdateTodoItemRequest.java
package com.example.todo.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Size;

/**
 * Request DTO for updating a todo item - all fields are optional (PATCH semantics).
 * Mirrors an UpdateTodoItemRequest in C#.
 */
public record UpdateTodoItemRequest(

        @Schema(example = "Buy groceries")
        @Size(max = 200, message = "Title must not exceed 200 characters")
        String title,

        @Schema(example = "Milk, eggs, bread")
        @Size(max = 2000, message = "Description must not exceed 2000 characters")
        String description,

        Boolean isCompleted
) {}
